metaci/release/tasks.py

```diff
@@ -310,6 +321,125 @@ def all_deps_satisfied(
     )
 
 
+PACKAGE_UPLOAD_TASK_PATHS = [
+    "cumulusci.tasks.create_package_version.CreatePackageVersion",
+    "cumulusci.tasks.salesforce.PackageUpload",
+]
+
+
+def get_package_ids_from_build(build: Build) -> Optional[DependencyGraphItem]:
+    """Traverse the Flow Tasks associated with this Build until we locate
+    a package upload task. Return a DependencyGraphItem with the package
+    details from its output, or None if not found."""
+    for build_flow in build.flows.all():
+        for flow_task in build_flow.tasks.all():
+            if flow_task.class_path in PACKAGE_UPLOAD_TASK_PATHS:
```

Comment:
Guessing a filter is more trouble than it's worth?

Reply:
I think I was going for deterministic behavior vis-a-vis iteration order. I don't feel strongly about it, though.

Reply:
I don't remember what was in my head. A filter would be nicer here; let me see if I can tweak that without breaking anything else.

Reply:
Done.